test/test.js

```diff
@@ -139,33 +145,68 @@ describe('rollup-plugin-babel', function() {
 	});
 
 	it('allows transform-runtime to be used instead of bundled helpers', () => {
-		let warnCalled = false;
+		const warnings = [];
 		return bundle(
 			'samples/runtime-helpers/main.js',
 			{ runtimeHelpers: true },
 			{},
 			{
-				onwarn: function(warning) {
-					assert.equal(warning.code, 'UNRESOLVED_IMPORT');
-					assert.equal(warning.source, '@babel/runtime/helpers/classCallCheck');
-					warnCalled = true;
+				onwarn(warning) {
+					warnings.push(warning.message);
 				},
 			},
 		).then(({ code }) => {
-			assert.ok(warnCalled, 'onwarn was never triggered about unresolved imports');
-			assert.ok(!~code.indexOf(HELPERS));
-		});
-	});
+			assert.deepStrictEqual(warnings, [
+				`'@babel/runtime/helpers/classCallCheck' is imported by samples${path.sep}runtime-helpers${path.sep}main.js, but could not be resolved – treating it as an external dependency`,
+			]);
+			assert.strictEqual(
+				code,
+				`'use strict';
 
-	it('allows transform-runtime to inject builtin version of helpers', () => {
-		return bundle('samples/runtime-helpers-esm/main.js', { runtimeHelpers: true }, {}, {}).then(({ code }) => {
-			assert.ok(!~code.indexOf(HELPERS));
+function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
+
+var _classCallCheck = _interopDefault(require('@babel/runtime/helpers/classCallCheck'));
+
+var Foo = function Foo() {
+  _classCallCheck(this, Foo);
+};
+
+module.exports = Foo;
+`,
+			);
 		});
 	});
 
 	it('allows transform-runtime to inject esm version of helpers', () => {
-		return bundle('samples/runtime-helpers-esm/main.js', { runtimeHelpers: true }, {}, {}).then(({ code }) => {
-			assert.ok(!~code.indexOf(HELPERS));
+		const warnings = [];
+		return bundle(
+			'samples/runtime-helpers-esm/main.js',
+			{ runtimeHelpers: true },
+			{},
+			{
+				onwarn(warning) {
+					warnings.push(warning.message);
+				},
+			},
+		).then(({ code }) => {
+			assert.deepStrictEqual(warnings, [
+				`'@babel/runtime/helpers/esm/classCallCheck' is imported by samples${path.sep}runtime-helpers-esm${path.sep}main.js, but could not be resolved – treating it as an external dependency`,
+			]);
+			assert.strictEqual(
+				code,
+				`'use strict';
+
+function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
+
+var _classCallCheck = _interopDefault(require('@babel/runtime/helpers/esm/classCallCheck'));
```

Comment:
this looks broken (even though has no impact on the test), esm helpers are being referenced in cjs output, could we fix this so it doesnt trip any future readers?

Reply:
This test could just use esm output and then it would be fine I think.